lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -340,11 +340,11 @@ defmodule Kernel.WarningTest do
     purge Sample
   end
 
-  test "unused guard" do
+  test "same result / always matches" do
```

Comment:
the thing was that this name didn't describe the second test. I have split them in two different tests. let me know what you think.